code/controllers/CMSPageHistoryController.php

```diff
@@ -358,7 +358,7 @@ public function ShowVersionForm($versionID = null) {
 	/**
 	 * @param int $versionID
 	 * @param int $otherVersionID
-	 * @return Form
+	 * @return mixed Form|boolean|SS_HTTPResponse
```

Comment:
@dhensby `Security::permissionFailure()` returns an instance of `SS_HTTPResponse` ergo, returning `Security::permissionFailure()` on line 381 will mean that `CompareVersionsForm()` can return one of:

* A boolean
* An instance of `SS_HTTPResponse`
* An instance of `Form`

..personally I think multiple / mixed return types is crazy, but let's not over think this. This is how the method has been for years :-)